packages/extension-chakra-storefront/src/types/config.ts

```diff
@@ -48,19 +49,56 @@ type EinsteinAPI = {
 // Indicates where a value should be placed in the URL.
 type UrlPlacement = 'path' | 'query_string' | 'none'
 
-type Pages = typeof import('../pages')
+// Default configuration type
+// should we keep string | string[] type here??
+type DefaultPageConfig = {
+    path: string
+}
+
+type CustomPageConfigs = {
+    ProductList: DefaultPageConfig & {
+        imageViewType: 'large'
+        selectedAttributedId: 'color'
+    }
+}
 
+// Combine inferred pages with specific configurations
+type PageConfigs = {
+    [K in keyof Pages]: K extends keyof CustomPageConfigs ? CustomPageConfigs[K] : DefaultPageConfig
+}
 /**
  * This defines how your extension can be configured in the user's project. Please update it to your specific needs!
  */
 export interface UserConfig extends ApplicationExtensionConfig {
     activeDataEnabled?: boolean // default = false
     commerceAPI: CommerceAPIConfig
+    categoryNav: {
+        defaultNavSsrDepth: number
+        defaultRootCategory: string | number
+    }
     defaultSite: Site['id']
+    defaultAppLocale: string
+    defaultSiteTitle: string
     einsteinAPI: EinsteinAPI
-    pages?: Record<keyof Pages, false | string | string[]> // if false, the page will not be shown
+    maxCacheAge: number
+    pages?: {
+        [K in keyof PageConfigs]: false | PageConfigs[K]
+    }
+    search: {
+        defaultLimitValues: number[]
+        defaultSearchParams: {
+            limit?: number
+            offset?: number
+            sort?: string
+            refine?: []
+        }
```

Comment:
Aside: to keep things simple, I'd say no need to worry about optional properties at this level. Just considering the _top-level_ properties like `search` is enough.

Besides, at this moment, the UserConfig type is not being used to validate the config.